File: spark_core/src/main/scala/com/test/business/WordCount.scala
package com.test.business

import com.test.common.BaseProgram
import org.apache.spark.sql.DataFrame

/**
  *
  * @param args (IDE 本地测试时，参数为：
        {
          \"input_dir\":\"spark_core/tmp/logs/test/1\",
          \"output_dir\":\"spark_core/tmp/out/test\",
          \"run_pattern\":\"local\"
        }
  */
object WordCount extends BaseProgram {

  private val dataFrame: DataFrame = readCsvToDataFrame(indexAndNameMap=Map(0 -> "name",2->"score"))

  dataFrame.createTempView("person")
  private val df2: DataFrame = sparkSession.sql("select * from person")
  df2.show()

  override def setSerializedClass():Unit = {
    val serializedClassArr:Array[Class[_]] = Array(classOf[Array[String]])
    sparkConf.registerKryoClasses(serializedClassArr)
  }
}

File: sparkML_python/src/com/test/business/template/wordCount.py
# -*- coding: UTF-8 -*-

import sys
from com.test.common.BaseProgram import BaseProgram


'''
    IDE 本地测试时，参数为：
        run_pattern=local
        app_name=wordCount
        input_dir=tmp/logs/test/1
        output_dir=tmp/out/test
'''
if __name__ == '__main__':

    program = BaseProgram(sys.argv)
    program.getSparkConf()
    program.getSparkSession()

    df = program.sparkSession().read.csv(program.paramDir['input_dir']).toDF("name", "age", "score")
    df.registerTempTable("Student")

    result = program.sparkSession().sql("select name,age from Student limit 2")
    result.show()

    result.write.csv(program.paramDir['output_dir'])
    sys.exit()
